doc/develop/modules.rst

```diff
@@ -761,6 +763,36 @@ are maintained in the module. For example:
     boards:
       - boards
 
+.. _modules-bin-blobs:
+
+Binary Blobs
+============
+
+Zephyr supports fetching and using :ref:`binary blobs <bin-blobs>`, and their
+metadata is contained entirely in :file:`zephyr/module.yml`. This is because
+a binary blob must always be associated with a Zephyr module, and thus the
+blob metadata belongs in the module's description itself.
+
+Binary blobs are fetched by :ref:`west <west-blobs>`, there is no support for
+fetching binary blobs when not using west.
+
+The ``blobs`` section in :file:`zephyr/module.yml` consists of a sequence of
+maps, each of which has the following entries:
+
+- ``path``: The path to the binary blob, relative to the :file:`zephyr/blobs/`
+  folder in the module repository
+- ``sha256``: `SHA-256 <https://en.wikipedia.org/wiki/SHA-2>`_ checksum of the
+  binary blob file
+- ``type``: The :ref:`type of binary blob <bin-blobs-types>`. Currently limited
+  to ``img`` or ``lib``
+- ``version``: A version string
```

Comment:
should we specify a format pattern here ?
or should it be free-form string.

CMake supports checking against `major[.minor[.patch[.tweak]]]` formatted versions.

So requiring that format, and not allowing e.g. `v1.2.3` will make checking / integration into the build system easier in the future.

> Component-wise integer version number comparison (version format is major[.minor[.patch[.tweak]]], omitted components are treated as zero). Any non-integer version component or non-integer trailing part of a version component effectively truncates the string at that point.

Ref: https://cmake.org/cmake/help/latest/command/if.html#version-comparisons

Reply:
Honestly I think this is overkill. Some vendors might prefer to have an incremental number here, we should not impose any particular versioning scheme on them, in my opinion.

Reply:
>  Some vendors might prefer to have an incremental number here

and even that is possible with my suggestion.
Notice that the only mandatory part is the `major` number, rest is optional.

but as we haven't planned for any specific use of this filed, i'm ok with not putting restrictions.
Just trying to avoid making things more complicated if we suddenly want to use this info.

Reply:
Fair enough. But in the name of flexibility, and given that it is very unlikely that we (Zephyr) ever use that field, I'd say let's remain flexible.

Reply:
No problem. We'll not know the right choice until the future is here.